### ISSUE TITLE
incorrect parsing of UTF-16 escape sequences

### ISSUE THREAD

**Issue by drsm** [nginx/njs#96]
```js
$ cat u16.js 
var s0 = 'фыва asdf 👍';
var s1 = '\u0444\u044b\u0432\u0430 asdf \ud83d\udc4d';
var s2 = JSON.parse('"\\u0444\\u044b\\u0432\\u0430 asdf \\ud83d\\udc4d"');
console.log(s0, '|', s1, '|', s2);
$ build/njs u16.js 
'фыва asdf 👍' '|' 'фыва asdf ������' '|' 'фыва asdf 👍'
```

**Comment by hongzhidao** [comment]
@drsm I'm courious how do you find this issue?
@xeioex @VBart I tried to port unicode_lowcase from UNIT, unfortunately it also failed.

**Comment by drsm** [comment]
@hongzhidao 
i've just made a mistake [there](http://mailman.nginx.org/pipermail/nginx-ru/2019-February/061882.html) :) .
then fix it, and hit the bug

**Comment by lexborisov** [comment]
@drsm 

Please, try [this patch](https://gist.github.com/78ac7d8c3ede83c4a2f2345ea878618c) to fix the problem.

**Comment by drsm** [comment]
@lexborisov 

Thank you for the patch, works fine for me `'\ud83d\udc4d \u{1f44d}'`!

**Comment by drsm** [comment]
@lexborisov 

seems that the patch also fixes #97 , would be nice to test it also.
